### PR TITLE
New module - AWS Glue connection

### PR DESCRIPTION
**Pull request by wimnat**
##### SUMMARY
New module for AWS Glue Connections

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
glue_connection

##### ANSIBLE VERSION
```
2.6
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/glue_connection.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/glue_connection.py]: New copyright: https://docs.ansible.com/ansible/2.5/dev_guide/developing_modules_documenting.html#copyright

File: lib/ansible/modules/cloud/amazon/glue_connection.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+physical_connection_requirements:
+    description: A dict of physical connection requirements, such as VPC and SecurityGroup,
+                 needed for making this connection successfully.
+    returned: when state is present
+    type: dict
+    sample: {'subnet-id':'subnet-aabbccddee'}
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, ec2_argument_spec
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/glue_connection.py]: AnsibleAWSModule adds the ec2_argument_spec by default, so you can remove this import and just specify the module arguments in a dict. See lower half of https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#porting-code-to-ansibleawsmodule.

File: lib/ansible/modules/cloud/amazon/glue_connection.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    if module.params.get("catalog_id") is not None:
+        params['CatalogId'] = module.params.get("catalog_id")
+    if module.params.get("description") is not None:
+        params['ConnectionInput']['Description'] = module.params.get("description")
+    if module.params.get("match_criteria") is not None:
+        params['ConnectionInput']['MatchCriteria'] = module.params.get("match_criteria")
+    if module.params.get("security_groups") is not None or module.params.get("subnet_id") is not None:
+        params['ConnectionInput']['PhysicalConnectionRequirements'] = dict()
+    if module.params.get("security_groups") is not None:
+        params['ConnectionInput']['PhysicalConnectionRequirements']['SecurityGroupIdList'] = module.params.get("security_groups")
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/glue_connection.py]: security_groups is documented as taking security group names or IDs. The docs could be updated or get_ec2_security_group_ids_from_names from module_utils.ec2 could be used.

File: lib/ansible/modules/cloud/amazon/glue_connection.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: glue_connection
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/glue_connection.py]: Maybe aws_glue_connection instead?

File: lib/ansible/modules/cloud/amazon/glue_connection.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+  connection_properties:
+    description:
+      - A dict of key-value pairs used as parameters for this connection.
+    required: true
+  connection_type:
+    description:
+      - The type of the connection. Currently, only JDBC is supported; SFTP is not supported.
+    required: false
+    default: JDBC
+    choices: [ 'JDBC', 'SFTP' ]
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/glue_connection.py]: Should SFTP be an option even provided at this point?

**Comment by s-hertel** [comment]
I added some integration tests as per https://github.com/ansible/community/blob/master/group-aws/integration.md#policy and will work on updating CI with the IAM permissions for those.

**Review by ryansb — APPROVED** [review]

**Pull request merged**